src/compiler/checker.ts

```diff
@@ -29331,60 +29331,71 @@ namespace ts {
-
-                        // It is an error to inherit an abstract member without implementing it or being declared abstract.
-                        // If there is no declaration for the derived class (as in the case of class expressions),
-                        // then the class cannot be declared abstract.
-                        if (baseDeclarationFlags & ModifierFlags.Abstract && (!derivedClassDecl || !hasModifier(derivedClassDecl, ModifierFlags.Abstract))) {
-                            if (derivedClassDecl.kind === SyntaxKind.ClassExpression) {
-                                error(derivedClassDecl, Diagnostics.Non_abstract_class_expression_does_not_implement_inherited_abstract_member_0_from_class_1,
-                                    symbolToString(baseProperty), typeToString(baseType));
-                            }
-                            else {
-                                error(derivedClassDecl, Diagnostics.Non_abstract_class_0_does_not_implement_inherited_abstract_member_1_from_class_2,
-                                    typeToString(type), symbolToString(baseProperty), typeToString(baseType));
+                // In order to resolve whether the inherited method was overridden in the base class or not,
+                // we compare the Symbols obtained. Since getTargetSymbol returns the symbol on the *uninstantiated*
+                // type declaration, derived and base resolve to the same symbol even in the case of generic classes.
+                if (derived === base) {
+                    // derived class inherits base without override/redeclaration
+
+                    const derivedClassDecl = getClassLikeDeclarationOfSymbol(type.symbol)!;
+
+                    // It is an error to inherit an abstract member without implementing it or being declared abstract.
+                    // If there is no declaration for the derived class (as in the case of class expressions),
+                    // then the class cannot be declared abstract.
+                    if (baseDeclarationFlags & ModifierFlags.Abstract && (!derivedClassDecl || !hasModifier(derivedClassDecl, ModifierFlags.Abstract))) {
+                        // getPropertyOfObjectType will return a single symbol even if the property can be found
+                        // on multiple declarations, so it’s possible for derived === base to be true even when
+                        // another base type introduced from declaration merging to contain the property as well,
+                        // which should satisfy the abstract member requirement.
+                        for (const otherBaseType of getBaseTypes(type)) {
+                            if (otherBaseType === baseType) continue;
+                            const baseSymbol = getPropertyOfObjectType(otherBaseType, base.escapedName);
+                            const derivedElsewhere = baseSymbol && getTargetSymbol(baseSymbol);

```

Comment:
Shouldn't this be
```suggestion
                            const derivedElsewhere = baseSymbol && getTargetSymbol(baseSymbol) || baseSymbol;
```

I just feel like `baseSymbol` might not always be an instantiated symbol (though it often is since a type reference in a base expression is pretty much always instantiated with a `this` type parameter, which forces symbol instantiation).

Reply:
I’m not 100% sure what that means... can you think of a test case that would show this? (Mostly so I can see it happen and internalize what you’re talking about.)

Reply:
@weswigham oh, I get what you’re saying now. But `getTargetSymbol` returns its argument if the symbol isn’t an instantiated symbol:

```ts
function getTargetSymbol(s: Symbol) {
  // if symbol is instantiated its flags are not copied from the 'target'
  // so we'll need to get back original 'target' symbol to work with correct set of flags
  return getCheckFlags(s) & CheckFlags.Instantiated ? (<TransientSymbol>s).target! : s;
}
``

Reply:
Oh, haaaaaah, so it already does that, duh.